use-cases/virgo/synthetic_data_gen/file_gen.py

```diff
@@ -0,0 +1,278 @@
+
+
+    original_width, original_height = qplot_array.shape
+
+    # Calculate the starting and ending indices for the subarray
+    start_x = max(center_x - square_size // 2, 0)
+    #print(start_x)
+    end_x = min(start_x + square_size, original_width)
+    #print(end_x)
+    start_y = max(center_y - square_size // 2, 0)
+    # print(start_y)
+    end_y = min(start_y + square_size, original_height)
+    # print(end_y)
+    # start_y=0
+    # end_y=square_size
+
+
+    # Adjust indices if needed to make sure the resulting subarray is (square_size X square_size)
+    if end_x - start_x < square_size:
+        diff_x = square_size - (end_x - start_x)
+        if end_x < original_width:
+            end_x += diff_x
+        else:
+            start_x -= diff_x
+    if end_y - start_y < square_size:
+        diff_y = square_size - (end_y - start_y)
+        if end_y < original_height:
+            end_y += diff_y
+        else:
+            start_y -= diff_y
+
+    subarray = qplot_array[start_x:end_x, start_y:end_y]
+    # print(subarray.shape)
+    # print(type(subarray))
+
+    return subarray
+
+
+
+def extract_peak_frequency(hq):
+    """
+    Calculates peak frequency (and relative time) of a given qplot
+
+    Input:
+    -hq (gwpy.Spectrgram) : Qtransform
+
+    Return:
+    -index_time (int) : time index in qtransform relative to peak frequency
+    -index_freq (int) : frequency index in qtransorm relative to peak frequency
+    """
+
+    #Calculate peak frequency, time and energy density
+    index_flat = np.argmax(hq.value)
+
+    # Convert the flattened index to 2D index
+    index_time, index_freq = np.unravel_index(index_flat, hq.shape)
+    peak_freq = hq.frequencies.value[index_freq]#/converting_factor_frequency
+    peak_value=np.max(hq.value)
+    peak_time = hq.times.value[index_time]
+
+    return index_time,index_freq
+
+
+def process_image(row,row_idx,channels,square_size):
+    """
+    Processes df's row to generate qplot images
+
+    Input:
+    - row (pd.Series) : row of TimeSeries Dataframe
+    - row_idx (int) : index relative to row in DataFrame
+    - channels (list): Name of columns of DataFrame
+    - square_size (int): Size in pixels of qplot image
+
+    Return:
+    df_row (DataFrame): Row containing qplot images as 2D np.array
+    """
+
+
+    res_list=[]
+    df_row=pd.DataFrame(columns=channels)
+    for i,channel in enumerate(channels):
+
+        qplot =row[channel].q_transform(frange=(10, 150))
+
+        #calculate peak frequency and time indices for main channel
+        if i==0:
+            index_time,index_freq= extract_peak_frequency(qplot)
+
+        # Convert the spectrogram to a NumPy array
+        qplot_array = qplot.value
+        qplot_array_cut= cut_image(qplot_array,index_freq,index_time,square_size)
+        df_row[channel]=[qplot_array_cut]
+
+    return df_row
+
+
+
+def generate_cut_image_dataset(df,channels,num_processes=20,square_size=128):
```

Comment:
Redefinition of function under `./use-cases/virgo/src/dataset.py::generate_cut_image_dataset`

Reply:
Do you think this whole file could be removed maybe? I don't see it being imported anywhere in the virgo folder.